helixbass/tsc-rust
# Parse + bind function declaration

In this PR:
- parse + bind a function declaration

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
function x(arg: number): boolean {
  return `${arg}`
}
```
then you should see `post-binding:` debug-logging where `x` is in the `SourceFile`'s `locals` `SymbolTable`, the only statement is a `FunctionDeclaration` whose `name` is an `Identifier` with `escaped_text` `"x"`, `parameters` contains a `ParameterDeclaration` whose `name` is an `Identifier` with `escaped_text` `"arg"` and `type_` is a `NumberKeyword` `KeywordTypeNode`, `type_` is a `BooleanKeyword` `KeywordTypeNode`, `body` is a `Block` containing a single `ReturnStatement` whose `expression` is a `TemplateExpression` (and `parent` bindings should be initialized). The `locals` `SymbolTable` on the `FunctionDeclaration` should contain `arg`

Based on `type-check-template-literal`
